AWS credentials file parsing crashes when it contains comments

Having a credentials file containing comments like:
```
# My comment
[default]
role_arn = ...
```

The parsing crashes and I get this stack trace:
```
smithy4s.aws.AwsCredentialsFileException: Unable to parse credentials file.
        at smithy4s.aws.AwsCredentialsFileException$.apply(AwsCredentialsFile.scala:27)
        at smithy4s.aws.AwsCredentialsFile$.processFileLines$$anonfun$2(AwsCredentialsFile.scala:117)
        at cats.syntax.EitherOps$.leftMap$extension(either.scala:193)
        at smithy4s.aws.AwsCredentialsFile$.processFileLines(AwsCredentialsFile.scala:117)
        at smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:46)
        at map @ smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:48)
        at flatMap @ smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:69)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at clear @ org.http4s.client.middleware.Retry$.retryLoop$1(Retry.scala:90)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at rethrow$extension @ fs2.Compiler$Target.compile$$anonfun$1(Compiler.scala:157)
        at run$ @ Main$.run(Main.scala:10)
        at get @ fs2.internal.Scope.openScope(Scope.scala:281)
        at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
        at flatMap @ fs2.Pull$.goCloseScope$1$$anonfun$1$$anonfun$3(Pull.scala:1215)
        at update @ fs2.internal.Scope.releaseChildScope(Scope.scala:227)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at smithy4s.aws.AwsCredentialsFile$.inProfile$1(AwsCredentialsFile.scala:88)
        at smithy4s.aws.AwsCredentialsFile$.lookingForProfile$1(AwsCredentialsFile.scala:106)
        at smithy4s.aws.AwsCredentialsFile$.processFileLines$$anonfun$1(AwsCredentialsFile.scala:114)
        at cats.syntax.EitherObjectOps$.catchNonFatal$extension(either.scala:391)
        at smithy4s.aws.AwsCredentialsFile$.processFileLines(AwsCredentialsFile.scala:114)
        at smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:46)
        at cats.effect.IOFiber.succeeded(IOFiber.scala:1172)
        at cats.effect.IOFiber.runLoop(IOFiber.scala:247)
        at cats.effect.IOFiber.asyncContinueFailedR(IOFiber.scala:1330)
        at cats.effect.IOFiber.run(IOFiber.scala:120)
        at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:555)
```

--- a/modules/aws/src/smithy4s/aws/AwsCredentialsFile.scala
+++ b/modules/aws/src/smithy4s/aws/AwsCredentialsFile.scala
@@ -91,6 +91,8 @@ object AwsCredentialsFile {
       }
     }
 
+    def withoutComments(line: String): String = line.replaceAll("#.+", "")
+
     def lookingForProfile(
         rest: List[String],
         data: Map[String, Map[String, String]]
@@ -111,7 +113,10 @@ object AwsCredentialsFile {
     }
     Either
       .catchNonFatal(
-        lookingForProfile(lines.filter(_.trim.nonEmpty), Map.empty)
+        lookingForProfile(
+          lines.map(withoutComments).filter(_.trim.nonEmpty),
+          Map.empty
+        )
       )
       .leftMap(ex =>
         AwsCredentialsFileException("Unable to parse credentials file.", ex)

--- a/modules/aws/test/src/smithy4s/aws/AwsCredentialsFileTest.scala
+++ b/modules/aws/test/src/smithy4s/aws/AwsCredentialsFileTest.scala
@@ -48,6 +48,22 @@ object AwsCredentialsFileTest extends FunSuite {
     }
   }
 
+  test("parse comments") {
+    expectRight(
+      AwsCredentialsFile.processFileLines(
+        asLines(
+          """|# A comment
+             |[default] # another comment
+             |aws_secret_access_key = sec
+             |aws_access_key_id     = key #yet another comment
+             |aws_session_token     = token""".stripMargin
+        )
+      )
+    ) { res =>
+      expect.same(Some(creds), res.default)
+    }
+  }
+
   test("find some other profile") {
     expectRight(
       AwsCredentialsFile.processFileLines(

